src/cloud_provider/service.rs

```diff
@@ -470,6 +470,41 @@ where
     Ok(())
 }
 
+pub fn scale_down_service(
+    target: &DeploymentTarget,
+    service: &impl Service,
+    replicas_count: usize,
+) -> Result<(), EngineError> {
+    let (kubernetes, environment) = match target {
+        DeploymentTarget::ManagedServices(_, _) => {
+            return Err(EngineError {
+                cause: EngineErrorCause::Internal,
+                scope: EngineErrorScope::Engine,
+                execution_id: service.context().execution_id().to_string(),
+                message: Some(format!("Cannot scale down managed service: {}", service.name_with_id())),
+            })
+        },
+        DeploymentTarget::SelfHosted(k, env) => {
+            (*k, *env)
+        },
+    };
+
+    let scaledown_ret = kubectl_exec_scale_replicas(
+        kubernetes.config_file_path()?,
+        kubernetes.cloud_provider().credentials_environment_variables(),
+        environment.namespace(),
+        ScalingKind::Deployment,
```

Comment:
Going to FIX: I suppose I need to check if it is a statefulset or a deployment, as app can be both